.github/workflows/netcdf-4.7.4.yml

```diff
@@ -0,0 +1,212 @@
+name: netcdf-4.7.4
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    env:
+      CC: gcc-9
+      FC: gfortran-9
+      CXX: g++-9
+    strategy:
+      matrix:
+        os: [macos-10.15, ubuntu-20.04]
+        compiler: [gcc-9]      
+
+    steps:
+    - name: cache-mpich
+      id: cache-mpich
+      uses: actions/cache@v2
+      with:
+        path: ~/mpich
+        key: mpich-${{ runner.os }}-3.3.2-1
+
+    - name: build-mpich
+      if: steps.cache-mpich.outputs.cache-hit != 'true'
+      run: |
+        if [[ ${{ matrix.os }} == "macos-10.15" ]]; then
+          wget http://www.mpich.org/static/downloads/3.3.2/mpich-3.3.2.tar.gz &> /dev/null
+          tar -xzf mpich-3.3.2.tar.gz
+          pushd mpich-3.3.2
+          ./configure --prefix=$HOME/mpich
+          make
+          sudo make install
+          popd
+        fi
+
+    - name: install-dependencies
+      run: |
+        echo "/home/runner/mpich/bin" >> $GITHUB_PATH
+        if [[ ${{ matrix.os }} == "ubuntu-20.04" ]]; then
+          sudo apt-get update
+          sudo apt-get install libmpich-dev
+          sudo apt-get install doxygen
+          sudo apt-get install pkg-config
+          sudo apt-get install libpng-dev
+          sudo apt-get install libjpeg-dev
+        elif [[ ${{ matrix.os }} == "macos-10.15" ]]; then
+          echo "$HOME/mpich/bin" >> $GITHUB_PATH
+          brew update
+          brew install doxygen
+          brew install mpich
+          brew install wget
+          if [[ ${{ matrix.compiler }} == "gcc-9" ]]; then
+            sudo ln -sf /usr/local/bin/gfortran-9 /usr/local/bin/gfortran
+          elif  [[ ${{ matrix.compiler }} == "gcc-10" ]]; then
+            sudo ln -sf /usr/local/bin/gfortran-10 /usr/local/bin/gfortran
+          fi
+        fi
+        python3 -m pip install gcovr
+
+    - name: cache-esmf
+      id: cache-esmf
+      uses: actions/cache@v2
+      with:
+        path: ~/esmf
+        key: esmf-${{ runner.os }}-8.0.1-2
+
+    - name: build-esmf
```

Comment:
Build NetCDF should come before building ESMF. If ESMF weren't cached it would fail building.

Reply:
Good catch! I have fixed, and pointed ESMF at the installed netCDF build.